src/server/Monte.AuthServer/Services/UserService.cs

```diff
@@ -28,11 +28,15 @@ public async Task<Result<UserDetails>> CreateUser(CreateUserRequest request, str
     {
         if (role == AuthConsts.Roles.MonteAdmin)
         {
-            if ((await _userManager.GetUsersInRoleAsync(AuthConsts.Roles.MonteAdmin)).Any())
+            var admins = await _userManager.GetUsersInRoleAsync(AuthConsts.Roles.MonteAdmin);
+            if (admins.Count > 0)
             {
-                return Result.Failure<UserDetails>(
-                    ErrorType.BadRequest,
-                    "Couldn't create admin, because an admin already exists in the system.");
+                var adminDetails = new UserDetails
+                {
+                    Id = admins[0].Id, Name = admins[0].UserName!, Role = AuthConsts.Roles.MonteAdmin
+                };
+
+                return Result.Success(adminDetails);
             }
```

Comment:
User creation is a POST request, which is not idempotent and this operation breaks the rule (I think).
Also someone is passing a CreateUserRequest object and gets a success while the provided data is not used, which may lead to confusion. I would return the admin details in some error status code (eg. 400, 405 or 409)

Reply:
admin creation is used internally at db setup, and has been removed from the controller

Reply:
Still, i don't like the idea that i can call the endpoint with the intent of creating an admin and it's giving me a OK result with the existing admin details without informing me, that only one admin is allowed in the system.

Reply:
there is no endpoint for admin creation

Reply:
i refactored the user creation code to be more clear

Reply:
Sorry, i completely forgot, that we had the user role hardcoded in the CreateUser endpoint. I thought that the endpoint could create any user. Again, sorry for the confusion.